api/v1alpha1/addon_types.go

```diff
@@ -134,40 +142,158 @@ func (ClusterAutoscaler) VariableSchema() clusterv1.VariableSchema {
 }
 
 type CSIProvider struct {
-	Name string `json:"name,omitempty"`
+	Name string `json:"name"`
+
+	// +optional
+	StorageClassConfig []StorageClassConfig `json:"storageClassConfig,omitempty"`
+
+	Strategy AddonStrategy `json:"strategy"`
+
+	// +optional
+	Credentials *corev1.SecretReference `json:"credentials,omitempty"`
+}
+
+type StorageClassConfig struct {
+	Name string `json:"name"`
+
+	// +optional
+	Parameters map[string]string `json:"parameters,omitempty"`
+
+	// +optional
+	ReclaimPolicy string `json:"reclaimPolicy,omitempty"`
+
+	// +optional
+	VolumeBindingMode string `json:"volumeBindingMode,omitempty"`
+}
+
+func (StorageClassConfig) VariableSchema() clusterv1.VariableSchema {
+	supportedReclaimPolicies := []string{
+		VolumeReclaimRecycle,
+		VolumeReclaimDelete,
+		VolumeReclaimRetain,
+	}
+	supportedBindingModes := []string{
+		VolumeBindingImmediate,
+		VolumeBindingWaitForFirstConsumer,
+	}
+	return clusterv1.VariableSchema{
+		OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+			Type: "object",
+			Properties: map[string]clusterv1.JSONSchemaProps{
+				"name": {
+					Type:        "string",
+					Description: "Name of storage class config.",
+				},
+				"parameters": {
+					Type:                   "object",
+					Description:            "Parameters passed into the storage class object.",
+					XPreserveUnknownFields: true,
+				},
+				"reclaimPolicy": {
+					Type: "string",
+					Enum: variables.MustMarshalValuesToEnumJSON(supportedReclaimPolicies...),
+				},
+				"volumeBindingMode": {
+					Type: "string",
+					Enum: variables.MustMarshalValuesToEnumJSON(supportedBindingModes...),
+				},
+			},
+		},
+	}
+}
+
+func (CSIProvider) VariableSchema() clusterv1.VariableSchema {
+	supportedCSIProviders := []string{CSIProviderAWSEBS, CSIProviderNutanix}
+	return clusterv1.VariableSchema{
+		OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+			Type: "object",
+			Properties: map[string]clusterv1.JSONSchemaProps{
+				"name": {
+					Description: "Name of the CSI Provider",
+					Type:        "string",
+					Enum: variables.MustMarshalValuesToEnumJSON(
+						supportedCSIProviders...),
+				},
+				"strategy": {
+					Description: "Addon strategy used to deploy the CSI provider to the workload cluster",
+					Type:        "string",
+					Enum: variables.MustMarshalValuesToEnumJSON(
+						AddonStrategyClusterResourceSet,
+						AddonStrategyHelmAddon,
+					),
+				},
+				"credentials": {
+					Type:        "object",
+					Description: "The reference to any secret used by the CSI Provider.",
+					Properties: map[string]clusterv1.JSONSchemaProps{
+						"name": {
+							Type: "string",
+						},
+						"namespace": {
+							Type: "string",
+						},
+					},
+				},
+				"storageClassConfig": {
+					Type: "array",
+					Items: &clusterv1.JSONSchemaProps{
+						Type:       "object",
+						Properties: StorageClassConfig{}.VariableSchema().OpenAPIV3Schema.Properties,
+					},
```

Comment:
Use the OpenAPI schema from `StorageClassConfig` as the schema for items

```suggestion
					Items: ptr.To(StorageClassConfig{}.VariableSchema().OpenAPIV3Schema},
```